akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -73,36 +76,14 @@ private[akka] class PublisherSink[In](val attributes: Attributes, shape: SinkSha
    * subscription a VirtualProcessor would perform (and it also saves overhead).
    */
   override def create(context: MaterializationContext): (AnyRef, Publisher[In]) = {
-    val proc = new VirtualPublisher[In]
+    val proc = new VirtualPublisher[In]()
```

Comment:
I'd argue exactly as your second sentence :) Seems like we have a good mix of both in the sources, so you can leave the change in place if you want.